cpp/src/arrow/io/io-file-test.cc

```diff
@@ -41,14 +41,34 @@ static bool FileExists(const std::string& path) {
   return std::ifstream(path.c_str()).good();
 }
 
+#if defined(_MSC_VER)
+void customInvalidParameterHandler(const wchar_t* expression,
+	const wchar_t* function,
+	const wchar_t* file,
+	unsigned int line,
+	uintptr_t pReserved)
+{
+	wprintf(L"Invalid parameter detected in function %s."
+		L" File: %s Line: %d\n", function, file, line);
+	wprintf(L"Expression: %s\n", expression);
+}
```

Comment:
See "Category A" licenses in https://www.apache.org/legal/resolved.html#category-a